components/cover/template.rst

```diff
@@ -92,20 +92,18 @@ with the ``cover.template.publish`` action.
     on_...:
       - cover.template.publish:
           id: template_cov
-          state: OPEN
+          position: OPEN
 
       # Templated
       - cover.template.publish:
           id: template_cov
-          state: !lambda 'return COVER_OPEN;'
+          position: !lambda 'return COVER_OPEN;'
 
 Configuration options:
 
 - **id** (**Required**, :ref:`config-id`): The ID of the template cover.
-- **state** (*Optional*, :ref:`templatable <config-templatable>`):
-  The state to publish. One of ``OPEN``, ``CLOSED``. If using a lambda, use ``COVER_OPEN`` or ``COVER_CLOSED``.
 - **position** (*Optional*, :ref:`templatable <config-templatable>`, float):
-  The position to publish, from 0 (CLOSED) to 1.0 (OPEN)
+  The position to publish, from 0 (CLOSED) to 1.0 (OPEN). If using a lambda, use ``COVER_OPEN`` or ``COVER_CLOSED``.
 - **tilt** (*Optional*, :ref:`templatable <config-templatable>`, float):
```

Comment:
Same here.

Reply:
This part perhaps needs a little clarification. The note afterward describes an alternative way that one could write a lambda. I believe it would be a reasonable assumption that `id(template_cov).state = COVER_OPEN;` would be valid, however, this throws a compile-time error.

> class esphome::template_::TemplateCover' has no member named 'state'

Reply:
Lambda/c++ variables do not always match the yaml configuration variables.
In YAML, `position` is a float, whereas `state` is a `string` `OPEN` or `CLOSED`

The removed field is in c++ `.state` and the `state` string in YAML is converted to 1.0 or 0.0 and then set to `position`

The `position` lambda for a template cover should not be doing `id(template_cov).state = COVER_OPEN;`, it should be `return`ing the position/state that the cover is in: `return COVER_OPEN;`

Reply:
If there are more instances where the C++ and YAML differ then if and how that should be documented has a broader impact that would be out of scope for this pull request.

As the documentation was correct in this instance (and the next) I will revert the change.